update join page with multi-hop limitation (#4674)
dbt-labs/docs.getdbt.com: website/docs/docs/build/join-logic.md

## Multi-hop joins

:::info

> MetricFlow currently supports multi-hop joins with up to two hops. This means we can render joins between three tables at most. 

Reviewer: small tweak here suggestion
```suggestion
MetricFlow can join three tables at most, supporting multi-hop joins with a limit of two hops.
```
